src/GraphQL/ObjectExtensions.cs

```diff
@@ -131,11 +131,25 @@ public static object ConvertValue(object value, Type fieldType)
         {
             if (value == null) return null;
 
-            if (fieldType == typeof(DateTime) && value is DateTime)
+            // exact type match
+            if ((fieldType == typeof(DateTime) && value is DateTime) ||
+                (fieldType == typeof(DateTimeOffset) && value is DateTimeOffset))
```

Comment:
Perhaps this can just be simplified to something like:

```
if (fieldType.IsInstanceOfType(value)) return value;
```

So if the given value is of the given type it just returns it.

Reply:
Yes, it should work for other types too.